src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,16 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+
+    @Override
+    public boolean test(Candidate candidate) {
+        return (candidate.getAge() >= 35 && candidate.getNationality().equals("Ukrainian")
+                && (Integer.parseInt(candidate.getPeriodsInUkr().split("-")[1])
+                        - Integer.parseInt(candidate.getPeriodsInUkr().split("-")[0]) > 10)
+                && candidate.isAllowedToVote());
```

Comment:
```suggestion
        String[] splitted = candidate.getPeriodsInUkr().split("-");
        return (candidate.getAge() >= REQUIRED_MIN_AGE && candidate.getNationality().equals(REUIRED_NATIONALITY)
                && (Integer.parseInt(splitted[PERIOD_TO_INDEX])
                        - Integer.parseInt(splitted[PERIOD_FROM_INDEX]) > REQUIRED_PERIOD_IN_COUNTRY)
                && candidate.isAllowedToVote());
```
lets move splitting the string to a variable, this way you will split data just once not twice
also lets create a few constants for indexes and required values